subsys/bluetooth/mesh/access.c

```diff
@@ -594,6 +595,19 @@ static bool model_has_dst(struct bt_mesh_model *mod, uint16_t dst)
 	return mod->elem_idx == 0;
 }
 
+bool bt_mesh_sub_addr_is_valid(uint16_t addr, uint16_t elem_addr)
+{
+	if (!BT_MESH_ADDR_IS_GROUP(addr) && !BT_MESH_ADDR_IS_FIXED_GROUP(addr)) {
+		return false;
+	}
+
+	if (BT_MESH_ADDR_IS_FIXED_GROUP(addr) && elem_addr == bt_mesh_primary_addr()) {
+		return false;
+	}
```

Comment:
Doesn't this contradict with this statement (from the comment in erratum):
```
3. If the feature is not supported, can anyone still subscribe models on the primary element to these fixed-group addresses?
YES
```

Reply:
Actually, shouldn't here `bt_mesh_fixed_group_match` function be used instead? My interpretation of the comment from erratum is that you can't assign a fixed group address to a node that doesn't support a feature:
```
2. If yes, does this subscription on a non-primary element is conditional to the given feature being supported? For example: can you use all-relay group address as a subscription address on a non-relay node?
NO
```
Or this should be clarified before making the change.

Reply:
I had the same question as you when I read the errata. 
I've already clarified it with @omkar3141 IRL since he is the errata author. `NO` for question 2 is for the first part of the question. No contradiction there. Probably @omkar3141 could give you more explanations.

`bt_mesh_fixed_group_match` cannot be used since it doesn't differentiate `BT_MESH_ADDR_ALL_NODES` fixed address from the others. This address is prohibited to use for the subscription.

Reply:
In any case, by reading answer on the 3rd question, there should not be any difference between primary and non-primary elements.